tests/test_cli_files.py

```diff
@@ -241,3 +241,101 @@ def test_move_file_running_workflow():
             )
             assert result.exit_code == 1
             assert message in result.output
+
+
+def test_list_files_filter():
+    """Test list workflow workspace files with filter."""
+    status_code = 200
+    response = {
+        "items": [
+            {
+                "last-modified": "2021-06-14T10:20:13",
+                "name": "data/names.txt",
+                "size": {"human_readable": "20 Bytes", "raw": 20},
+            },
+        ]
+    }
+    env = {"REANA_SERVER_URL": "localhost"}
+    mock_http_response, mock_response = Mock(), Mock()
+    mock_http_response.status_code = status_code
+    mock_response = response
+    reana_token = "000000"
+    runner = CliRunner(env=env)
+    with runner.isolation():
+        with patch(
+            "reana_client.api.client.current_rs_api_client",
+            make_mock_api_client("reana-server")(mock_response, mock_http_response),
+        ):
+            result = runner.invoke(
+                cli,
+                [
+                    "ls",
+                    "-t",
+                    reana_token,
+                    "--workflow",
+                    "mytest.1",
+                    "--filter",
+                    "name=names",
```

Comment:
Beware about exact string matching vs substring matching.

For `list`, we match partially:

```console
$ reana-client list --filter name=w 
NAME                           RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
helloworld-serial-kubernetes   1            2021-06-14T14:33:41   2021-06-14T14:33:54   2021-06-14T14:34:01   finished
helloworld-cwl-kubernetes      1            2021-06-14T14:33:14   2021-06-14T14:33:28   2021-06-14T14:33:36   finished
helloworld-yadage-kubernetes   1            2021-06-14T14:32:33   2021-06-14T14:32:47   2021-06-14T14:33:09   finished
```

For `ls`, we match exactly:

```console
$ reana-client ls -w helloworld-cwl-kubernetes --filter name=workflow.json
NAME            SIZE   LAST-MODIFIED      
workflow.json   1287   2021-06-14T14:33:28

$ reana-client ls -w helloworld-cwl-kubernetes --filter name=w            
NAME   SIZE   LAST-MODIFIED
```

It would be good to harmonise this.  

Can you please fix the behaviour and add a test case? 

(Note that this was already mentioned in #521 issue description.)

Reply:
I just checked with my local setup again. Sub string match is working here. I also ran the `helloworld-cwl` example.

<img width="813" alt="Screenshot 2021-06-14 at 4 54 19 PM" src="https://user-images.githubusercontent.com/24358501/121912805-2acb7300-cd31-11eb-8c6f-9fc844300a5f.png">

Reply:
Hmm, do you have some uncommitted local code changes perhaps? Because it does not seem to work for me on a clean build:

```console
$ reana-dev git-clean -c ALL
$ reana-dev git-fetch -c ALL
$ reana-dev git-status -c ALL | grep -v '@ master'
reana-workflow-controller @ pr-384 @ f573f58 api: add search param in /workspace
reana-client @ pr-525 @ 99993c3 cli: add --filter option in ls command
reana-commons @ pr-268 @ 4554153 api-schema: add search param in /workspace
reana-server @ pr-372 @ a6ec996 api: add search param in get_files
$ reana-dev run-ci -m /var/reana:/var/reana --exclude-components=r-a-krb5,r-a-vomsproxy -c r-d-helloworld --admin-email xxx --admin-password xxx
$ eval $(reana-dev client-setup-environment)
$ reana-client ls -w helloworld-cwl-kubernetes                   
NAME                                                                        SIZE   LAST-MODIFIED      
workflow.json                                                               1287   2021-06-14T15:36:46
inputs.json                                                                 145    2021-06-14T15:36:46
code/helloworld.py                                                          3253   2021-06-14T15:36:33
data/names.txt                                                              20     2021-06-14T15:36:34
outputs/greetings.txt                                                       34     2021-06-14T15:36:49
workflow/cwl/helloworld-job.yml                                             122    2021-06-14T15:36:34
workflow/cwl/helloworld.cwl                                                 867    2021-06-14T15:36:34
workflow/cwl/helloworld.tool                                                541    2021-06-14T15:36:34
workflow/cwl/helloworld-slurmcern.cwl                                       965    2021-06-14T15:36:34
workflow/cwl/helloworld-htcondorcern.cwl                                    1007   2021-06-14T15:36:34
cwl/docker_stagedir/stg29dd9d6b-dfc7-4744-ae78-6569cf7f396c/helloworld.py   3253   2021-06-14T15:36:33
cwl/docker_stagedir/stg51f98c0e-9bfd-441e-8759-b5dac5712f3b/names.txt       20     2021-06-14T15:36:34
cwl/docker_outdir/results/greetings.txt                                     34     2021-06-14T15:36:49
$ reana-client ls -w helloworld-cwl-kubernetes --filter name=inputs.json
NAME          SIZE   LAST-MODIFIED      
inputs.json   145    2021-06-14T15:36:46
$ reana-client ls -w helloworld-cwl-kubernetes --filter name=inputs     
NAME   SIZE   LAST-MODIFIED
```

```

Reply:
This works after r-w-controller PR update :+1: